lib/src/extend/functions.dart

```diff
@@ -108,16 +107,44 @@ List<ComplexSelector>? unifyComplex(
 ///
 /// The [span] will be used for the new unified selector.
 ///
+/// This function ensures that the relative order of pseudo-classes (`:`) and
+/// pseudo-elements (`::`) within each input selector is preserved in the
+/// resulting combined selector.
+///
+/// This function enforces a general rule that pseudo-classes (`:`) should come
```

Comment:
Is this rule just applicable to pseudo-classes, or is it be applicable to _all_ simple selectors? It doesn't look like this function has any special handling for pseudo selectors.

Reply:
done, I added more test cases to catch bugs like.. `::foo.x` and `.x` which resulted in `.x::foo.x` 😅. It should now be producing the correct combined selector